### PR TITLE
can_medium -> can_food for 250ml items

### LINKED ISSUE #71946: The problem with canned beans
### Describe the bug

It's based on a theme: #71938

A vagrant and a hobo start with a sealed medium tin contains of beans. The problem is the number of beans.

There is only one bean (0.25 liter) in an medium can (0.5 liter) designed for two beans.

As a result, the game sees a half-empty container and tries to place items in it, but it is sealed. This leads to the problem described in #71938

### Attach save file

N/A

### Steps to reproduce

1. Select homeless or vagrant at character creation
2. Look at the can in your inventory and make sure it contains one bean.
3. Complete the character creation.
4. Remove all clothing except for the pants that contain the can
5. Start collecting trees (pines, oaks, birches, and so on)
6. When the pockets are full. Note the constant question of what to do with the item
7. Throw the can away
8. Continue harvesting trees. Note that the game does not ask you what to do with the result of the collection. It remains on the ground. 

### Expected behavior

N/A

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.18362 (21H1)
- Game Version: e5fd033 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

Vagabond and Hobo

https://github.com/CleverRaven/Cataclysm-DDA/blob/d6c98a1fd9ddd2b72fa459a1ec9c8f4abc502dce/data/json/professions.json#L605

https://github.com/CleverRaven/Cataclysm-DDA/blob/d6c98a1fd9ddd2b72fa459a1ec9c8f4abc502dce/data/json/professions.json#L2206

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change
Fixes #71946

#### Describe the solution
Moves 250ml foodstuffs currently defaulting into 500ml `can_medium`s into 250ml `can_food`s instead, the autogenned item groups in collections_domestic.json will still work to spawn two in a `can_medium` where desired.

#### Describe alternatives you've considered
Deleting default containers altogether bc they suck

#### Testing
image

### CODE REVIEW

**Comment by SurFlurer** [comment]
You can as well migrate the two appearances of `can_bean` noted in #71946 to group `can_beans_can_medium_2`, as that was the intended form before charge removal.

File: data/json/professions.json [hunk 1]
```diff
@@ -540,8 +540,8 @@
   },
   {
     "type": "profession_item_substitutions",
```
Comment (Maleclypse) [inline data/json/professions.json:542]: Can profession item substitions not use `group`?

Comment (Procyonae) [reply data/json/professions.json:542]: They can to some extent but not like this, I'll just switch it for 2 small cans instead bc it seems pretty overkill to add C++ for this

Comment (Procyonae) [reply data/json/professions.json:542]: Nvm that doesn't work either

**Comment by I-am-Erk** [comment]
getting some comestible test failures now

**Pull request closed without merging**